openfga/v1/openfga_service.proto

```diff
@@ -326,6 +327,73 @@ service OpenFGAService {
     };
   }
 
+  rpc BatchCheck(BatchCheckRequest) returns (BatchCheckResponse) {
+    option (google.api.http) = {
+      post: "/stores/{store_id}/batch-check"
+      body: "*"
+    };
+
+    option (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_operation) = {
+      summary: "Send a list of `check` operations in a single request"
+      tags: ["Relationship Queries"]
+      operation_id: "BatchCheck"
+      description:
+        "The `BatchCheck` API functions nearly identically to `Check`, but instead of checking a single "
+        "user-object relationship BatchCheck accepts a list of relationships to check and returns a map containing a response "
+        "for each check it received.\n\n"
+        "Within each `BatchCheckItem`, this endpoint requires a `correlation_id` in the form of a string containing only "
+        "alphanumeric characters or hyphens with a maximum length of 36 characters. This `correlation_id` is used to map the result "
+        "of each check to the item which was checked, so it must be unique for each item in the batch. "
+        "We recommend using a UUID or ULID as the `correlation_id`, but you can use whatever unique identifier you need as long "
+        " as it matches this regex pattern: `^[\\w\\d-]{1,36}$`\n\n"
+        "For more details on how `Check` functions, see the docs for `/check`.\n\n"
+        "### Examples\n"
+        "#### A BatchCheckRequest\n"
+        "```json\n"
+        "{\n"
+        "  \"checks\": [\n"
+        "     {\n"
+        "       \"tuple_key\": {\n"
+        "         \"object\": \"document:2021-budget\"\n"
+        "         \"relation\": \"reader\",\n"
+        "         \"user\": \"user:anne\",\n"
+        "       },\n"
+        "       \"contextual_tuples\": {...}\n"
+        "       \"context\": {}\n"
+        "       \"correlation_id\": \"01JA8PM3QM7VBPGB8KMPK8SBD5\"\n"
```

Comment:
Should we use a UUID in this example if we're using it in the examples on request/response?

Reply:
I think it's ok to have both, as long as each example is internally consistent